substrate/primitives/metadata-ir/src/unstable.rs

```diff
@@ -0,0 +1,212 @@
+		}
+	}
+}
+
+impl From<RuntimeApiMethodParamMetadataIR> for RuntimeApiMethodParamMetadata {
+	fn from(ir: RuntimeApiMethodParamMetadataIR) -> Self {
+		RuntimeApiMethodParamMetadata { name: ir.name, ty: ir.ty }
+	}
+}
+
+impl From<PalletMetadataIR> for PalletMetadata {
+	fn from(ir: PalletMetadataIR) -> Self {
+		PalletMetadata {
+			name: ir.name,
+			storage: ir.storage.map(Into::into),
+			calls: ir.calls.map(Into::into),
+			event: ir.event.map(Into::into),
+			constants: ir.constants.into_iter().map(Into::into).collect(),
+			error: ir.error.map(Into::into),
+			index: ir.index,
+			docs: ir.docs,
+			associated_types: ir.associated_types.into_iter().map(Into::into).collect(),
+			deprecation_info: ir.deprecation_info.into(),
+		}
+	}
+}
+
+impl From<PalletStorageMetadataIR> for PalletStorageMetadata {
+	fn from(ir: PalletStorageMetadataIR) -> Self {
+		PalletStorageMetadata {
+			prefix: ir.prefix,
+			entries: ir.entries.into_iter().map(Into::into).collect(),
+		}
+	}
+}
+
+impl From<StorageEntryMetadataIR> for StorageEntryMetadata {
+	fn from(ir: StorageEntryMetadataIR) -> Self {
+		StorageEntryMetadata {
+			name: ir.name,
+			modifier: ir.modifier.into(),
+			ty: ir.ty.into(),
+			default: ir.default,
+			docs: ir.docs,
+			deprecation_info: ir.deprecation_info.into(),
+		}
+	}
+}
+
+impl From<PalletAssociatedTypeMetadataIR> for PalletAssociatedTypeMetadata {
+	fn from(ir: PalletAssociatedTypeMetadataIR) -> Self {
+		PalletAssociatedTypeMetadata { name: ir.name, ty: ir.ty, docs: ir.docs }
+	}
+}
+
+impl From<PalletErrorMetadataIR> for PalletErrorMetadata {
+	fn from(ir: PalletErrorMetadataIR) -> Self {
+		PalletErrorMetadata { ty: ir.ty, deprecation_info: ir.deprecation_info.into() }
+	}
+}
+
+impl From<PalletEventMetadataIR> for PalletEventMetadata {
+	fn from(ir: PalletEventMetadataIR) -> Self {
+		PalletEventMetadata { ty: ir.ty, deprecation_info: ir.deprecation_info.into() }
+	}
+}
+
+impl From<PalletCallMetadataIR> for PalletCallMetadata {
+	fn from(ir: PalletCallMetadataIR) -> Self {
+		PalletCallMetadata { ty: ir.ty, deprecation_info: ir.deprecation_info.into() }
+	}
+}
+
+impl From<PalletConstantMetadataIR> for PalletConstantMetadata {
+	fn from(ir: PalletConstantMetadataIR) -> Self {
+		PalletConstantMetadata {
+			name: ir.name,
+			ty: ir.ty,
+			value: ir.value,
+			docs: ir.docs,
+			deprecation_info: ir.deprecation_info.into(),
+		}
+	}
+}
+
+impl From<TransactionExtensionMetadataIR> for TransactionExtensionMetadata {
+	fn from(ir: TransactionExtensionMetadataIR) -> Self {
+		TransactionExtensionMetadata {
+			identifier: ir.identifier,
+			ty: ir.ty,
+			additional_signed: ir.implicit,
+		}
+	}
+}
+
+impl From<ExtrinsicMetadataIR> for ExtrinsicMetadata {
+	fn from(ir: ExtrinsicMetadataIR) -> Self {
+		ExtrinsicMetadata {
+			versions: ir.versions.into_iter().map(Into::into).collect(),
+			address_ty: ir.address_ty,
+			call_ty: ir.call_ty,
+			signature_ty: ir.signature_ty,
+			extra_ty: ir.extra_ty,
```

Comment:
Just a reminder that we should remove this type from V16; can't remember if there is an issue for this already :)